### PR TITLE
Add marketing extensions task to task list

### LINKED ISSUE #7313: Add task to install recommended marketing extensions
### User story

As a merchant,
I want to be prompted to set up marketing tool as part of my store set up,
so that I am aware of the marketing tools that can help my store be successful.

### Description

A new _Set up marketing tools_ task in the Home Screen's _Get ready to start selling_ task list will be added.

### Design

Screen Shot 2021-07-08 at 3 04 57 PM

More details and design in pbIJXs-Ta-p2

### Acceptance criteria

- [ ] Introduce a new task in the setup checklist called "Set up marketing tools" (the title is tbd) and with the following description: "Add recommended marketing tools to reach new customers and grow your business"
- [ ] This new task should be placed after the tax task and before the personalize store task
- [ ] When the user clicks in the marketing task present a "Recommended marketing extensions card" with the following extensions: Mailpoet, Google Ads, Mailchimp and Creative Mail. Here's the copy of each extension:

Mailpoet
Create and send purchase follow-up emails, newsletters, and promotional campaigns straight from your dashboard.

Google Listings and Ads
Get in front of shoppers and drive traffic so you can grow your business with Smart Shopping Campaigns and free listings.

Mailchimp
Send targeted campaigns, recover abandoned carts and much more with Mailchimp.

Creative Mail
Create on-brand store campaigns, fast email promotions and customer retargeting with Creative Mail.

- [ ] We should display the following copy below the card: The extension will be installed and activated for you when you click "Install".
- [ ] When the user clicks "Install", we should automatically install the extensions from wp.org
https://wordpress.org/plugins/mailpoet/
https://wordpress.org/plugins/google-listings-and-ads/
https://wordpress.org/plugins/mailchimp-for-woocommerce/
https://wordpress.org/plugins/creative-mail-by-constant-contact/

### Tasks

- [ ] Creating a new task shown in the task list
- [ ] Record Tracks event `wcadmin_tasklist_click` with a new value called `marketing` in the `task_name` prop when the task is clicked
- [ ] Record Tracks event `wcadmin_task_view` with a new value called `marketing` in the `task_name` prop when the task page is shown
- [ ] Fetching and displaying categories from the free extensions REST endpoint `/wc-admin/onboarding/free-extensions`
- [ ] Handling installation of fetched plugins on clicking _Install_ and showing a success notice (and handling failure as well)
- [ ] Record Tracks event `wcadmin_tasklist_marketing_install` on extension installation with the following props:
    - `selected_extension` : the name of the extension that was selected to be installed
    - `installed_extensions` : an array showing the extensions that were installed when the user clicked to install one extension (TODO: follow up with Pedro to see what was intended)

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #7313 

Adds the marketing task to install marketing extensions from the free extensions REST API.

Note that this PR does not cover:
* Separating into multiple lists
* Task completion
* Changes to the free extensions REST API
* Changes to extensions in the profiler

### Screenshots

Screen Shot 2021-07-20 at 1 38 20 PM

### Detailed test instructions:

1. Checkout this WCCOM PR - https://github.com/Automattic/woocommerce.com/pull/10826
2. In `src/Features/RemoteFreeExtensions/DataSourcePoller.php` update the data source to `https://woocommerce.test/wp-json/wccom/obw-free-extensions/2.0/extensions.json`
3. Visit the marketing task in the task list
4. Note that buttons show `Activate|Manage|Install` depending on the plugin's state
5. Check that the buttons work as expected
6. Check that the "basics" plugin list is not shown
6. In your console, run `localStorage.setItem( 'debug', 'wc-admin:*' );`
7. Check that the `wcadmin_tasklist_marketing_install` event fires with the installed plugins and selected installed/activate plugin.

### CODE REVIEW

**Review by louwie17 — COMMENTED** [review]
Sorry if I stole this review from you @joelclimbsthings, as I noticed you assigned yourself yesterday.
Overall the code looks good. In order to get it to work locally I had to merge in the changes of #7391 into this PR while having this 10826-gh-Automattic/woocommerce.com PR branch loaded for my WC.com instance.

I only noticed one visual thing when testing. The **Recommended marketing extensions** header does not look like your screenshot (not sure why - maybe a WP 5.8 thing):
Screen Shot 2021-07-27 at 9 37 28 AM

Also I assume the WC.com PR would have to be merged in at the same time both of these PRs go in? Given these changes are already live, we might want to add a API version to the WC.com changes?

When this is all done, this will probably require a demo for Pedro & Elizabeth before merging.

**Comment by joshuatf** [comment]
Thanks for reviewing, @louwie17!

>I only noticed one visual thing when testing. The Recommended marketing extensions header does not look like your screenshot (not sure why - maybe a WP 5.8 thing):

I think this is related to breaking changes in the `Text` component in GB (I'm guessing you have the new plugin active).  Disabling the plugin should fix the styles, but I also added fixes for this in https://github.com/woocommerce/woocommerce-admin/pull/7419.  I can pull those changes over here if you prefer.

>Also I assume the WC.com PR would have to be merged in at the same time both of these PRs go in? Given these changes are already live, we might want to add a API version to the WC.com changes?

I don't think it has to be merged at the same time, but ideally within this cycle and by the next release at the very latest.

**Review by louwie17 — COMMENTED** [review]
Nice work on this @joshuatf, and thanks for the updates and updating the wc.com PR as well 👌 It tested well, I just had to rebase to make sure the changes of #7391 where present.
The header rendered incorrectly still, which the changes in #7419 fixed.
You got my approval after the completed state is added and either #7419 is merged or those css changes are added 😄

File: client/task-list/tasks/Marketing/index.tsx
Comment (joshuatf) [reply client/task-list/tasks/Marketing/index.tsx:92]: Thanks!

File: client/task-list/tasks/Marketing/index.tsx
Comment (joshuatf) [reply client/task-list/tasks/Marketing/index.tsx:79]: See https://github.com/woocommerce/woocommerce-admin/pull/7420

File: client/task-list/tasks.js
Comment (joshuatf) [reply client/task-list/tasks.js:347]: I think this is fine to merge in given we already have the update ready in https://github.com/woocommerce/woocommerce-admin/pull/7420 and can improve iteratively.  I'd like to get this merged to unblock other PRs.

File: client/task-list/tasks/Marketing/index.tsx
Comment (louwie17) [reply client/task-list/tasks/Marketing/index.tsx:79]: Awesome 😄

File: client/task-list/tasks.js
Comment (louwie17) [reply client/task-list/tasks.js:347]: Sounds good, should be good to approve then aside from the css changes.

**Review by louwie17 — APPROVED** [review]
Approving this, given the answers in the above replies. And that the `Marketing.scss` will be updated.

**Pull request merged**
